el/index.js

```diff
@@ -93,6 +106,26 @@ module.exports = yeoman.generators.Base.extend({
       this.writeFileFromString(file, 'app/elements/elements.html');
     }
 
+    if (this.testType && this.testType !== 'None') {
+      var testDir = 'app/test';	
+      
+      if (this.testType === 'TDD') {
+        this.template(path.join(__dirname, 'templates/test/_tdd.html'), path.join(testDir, this.elementName+'-basic.html'));
+      } else if (this.testType === 'BDD') {
+        this.template(path.join(__dirname, 'templates/test/_bdd.html'), path.join(testDir, this.elementName+'-basic.html'));
+      }
+
+      // Open index.html, locate where to insert text, insert ", x-foo.html" into the array of components to test
+      var indexFileName = 'app/test/index.html';
+      var origionalFile = this.readFileAsString(indexFileName);
+      var regex = /WCT\.loadSuites\(\[([^\]]*)/;
+      var match = regex.exec(origionalFile);
+      var indexOfInsertion = match.index + match[0].length;
+      var comma = (match[1].length === 0) ? '' : ', ';
+      var newFile = origionalFile.slice(0, indexOfInsertion) + comma + '\'' + this.elementName + '-basic.html\'' + origionalFile.slice(indexOfInsertion);
```

Comment:
Instead of slicing the array string, would it be easier to extract the entire array, run it through JSON.parse, push in the new value, run it through JSON.stringify and write that back?